libnetwork/network/interface.go

```diff
@@ -137,24 +137,30 @@ func defaultNetworkBackend(store storage.Store, conf *config.Config) (backend ty
 	if err != nil {
 		return "", err
 	}
-	if len(cons) == 0 {
-		imgs, err := store.Images()
-		if err != nil {
-			return "", err
-		}
-		if len(imgs) == 0 {
+	if len(cons) != 0 {
+		return types.CNI, nil
+	}
+
+	imgs, err := store.Images()
+	if err != nil {
+		return "", err
+	}
+	// ignore ReadOnly images
+	for _, i := range imgs {
+		if !i.ReadOnly {
```

Comment:
if the image is not readonly we should use cni, so there is no point in checking the networks

Reply:
This is not what the current code did, but I am fine changing it.

Reply:
No you still have to check networks, my point is that the place you check them was incorrect.

You check when there is a image check networks, this is wrong.
It must be when there is no image (or now only read only images) then check networks